docs/guides/documentation/releases/gh_aggregate_release_notes.py

```diff
@@ -0,0 +1,99 @@
+import requests
+import yaml
+import argparse
+from urllib.parse import urlparse
+from time import sleep
+from tqdm import tqdm
+import random
+
+def read_config(file_path):
+    """
+    Reads the YAML configuration file and returns its contents.
+    
+    Args:
+        file_path (str): The path to the YAML configuration file.
+        
+    Returns:
+        dict: The contents of the YAML file if successful, None otherwise.
+    """
+    with open(file_path, 'r') as stream:
+        try:
+            return yaml.safe_load(stream)
+        except yaml.YAMLError as exc:
+            print(exc)
+            return None
+
+def get_release_notes(url, github_token):
+    """
+    Extracts the owner, repo, and tag from the given GitHub URL and retrieves the release notes using the GitHub API.
+    Implements retries with exponential backoff and jitter for handling API rate limits and network issues.
+    
+    Args:
+        url (str): The URL of the GitHub release tag page.
+        github_token (str): The GitHub token used for authentication.
+    
+    Returns:
+        str: The release notes text if successful, empty string otherwise.
+    """
+    parsed_url = urlparse(url)
+    hostname = parsed_url.hostname
+    path_parts = parsed_url.path.split('/')
+    owner, repo, tag = path_parts[1], path_parts[2], path_parts[-1]
+
+    # Be agnostic to GitHub.com or GitHub Enterprise repo URLs
+    api_url_prefix = f"https://api.github.com/repos/{owner}/{repo}" if hostname == "github.com" else f"https://{hostname}/api/v3/repos/{owner}/{repo}"
+    api_url = f"{api_url_prefix}/releases/tags/{tag}"
+    headers = {
+        "Authorization": f"token {github_token}",
+        "Accept": "application/vnd.github.v3+json"
+    }
+
+    max_retries = 5
+    retry_num = 0
+    backoff_factor = 2
+    while retry_num < max_retries:
+        response = requests.get(api_url, headers=headers)
+        if response.status_code == 200:
+            return response.json().get("body", "")
+        else:
+            print(f"Error fetching release notes for {url}: {response.status_code}. Retrying...")
```

Comment:
Maybe move the `retry_num += 1` up and add an `if retry_num < max_retries` to condition the sleep? The "edge" case is that GitHub gives non-200 every single time, but we end up doing a final sleep for 2 × 2⁴ + [0.0..1.0] seconds—up to 33 seconds—to then just abort without a final attempt.

In other words, change from `attempt…sleep…attempt…sleep…attempt…sleep…attempt…sleep…abort` → `attempt…sleep…attempt…sleep…attempt…sleep…attempt…abort` 🤷 

Not a huge deal but I've been known to be rather pedantic 😂

Reply:
Let me look into this thanks @nutjob4life